lib/widgets/fields/alignment_field.dart

```diff
@@ -0,0 +1,186 @@
+import 'package:flutter/material.dart';
+
+enum AlignmentOption {
+  topLeft,
+  topCenter,
+  topRight,
+  centerLeft,
+  center,
+  centerRight,
+  bottomLeft,
+  bottomCenter,
+  bottomRight,
+}
+
+class AlignmentField extends StatefulWidget {
+  final void Function(Alignment) onChanged;
+
+  const AlignmentField({
+    Key? key,
+    required this.onChanged,
+  }) : super(key: key);
+
+  @override
+  _AlignmentFieldState createState() => _AlignmentFieldState();
+}
+
+class _AlignmentFieldState extends State<AlignmentField> {
+
+  AlignmentOption selectedOption = AlignmentOption.center;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Container(
+          width: 318,
+          decoration: BoxDecoration(
+            color: const Color(0xFF211F26),
+            borderRadius: BorderRadius.circular(8),
+            border: Border.all(
+              color: Colors.white,
+              width: 1,
+            ),
+          ),
+          child: Column(
+            children: List.generate(3, (row) {
+              return Row(
+                children: List.generate(3, (col) {
+                  AlignmentOption option = AlignmentOption.values[row * 3 + col];
+                  return AlignmentFieldItem(
+                    option: option,
+                    onChanged: (alignment) {
+                      setState(() {
+                        widget.onChanged(alignment);
+                        if (selectedOption != option) {
+                          selectedOption = option; // Deselect if already selected.
+                        }
+                      });
+                    },
+                    isSelected: selectedOption == option,
+                  );
+                }),
+              );
+            }),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class AlignmentFieldItem extends StatefulWidget {
+
+  final AlignmentOption option;
+  final void Function(Alignment) onChanged;
+  final bool isSelected;
+
+  const AlignmentFieldItem({
+    super.key,
+    required this.onChanged,
+    required this.isSelected,
+    required this.option,
+  });
+
+  @override
+  State<AlignmentFieldItem> createState() => _AlignmentFieldItemState();
+}
+
+class _AlignmentFieldItemState extends State<AlignmentFieldItem> {
+
+  BorderRadius? borderRadius;
+  String text = "";
+  Alignment? alignment;
+
+  AlignmentOption? onHoverOption;
+  bool isHover = false;
+
+  @override
+  Widget build(BuildContext context) {
+    switch (widget.option) {
```

Comment:
Always try to keep build methods as short as possible, here, instead of directly using the switch inside build, we can have an anonymous extension methods that provide text, alignment and borderRadius for different types of options.

Reply:
An anonymous extension is an unnamed extension that cannot be used outside of this file.